packages/typespec-client-generator-core/test/decorators/mark-as-pageable.test.ts

```diff
@@ -0,0 +1,453 @@
+  await basicRunner.compile(`
+      @service
+      namespace TestService {
+        model ItemListResult {
+          @pageItems
+          items: Item[];
+          @nextLink
+          nextLink?: string;
+        }
+
+        model Item {
+          id: string;
+          name: string;
+        }
+
+        @Azure.ClientGenerator.Core.Legacy.markAsPageable("csharp")
+        @route("/items")
+        @get
+        op listItems(): ItemListResult;
+      }
+    `);
+
+  const methods = basicRunner.context.sdkPackage.clients[0].methods;
+  strictEqual(methods.length, 1);
+
+  const method = methods[0];
+  strictEqual(method.kind, "paging");
+  strictEqual(method.name, "listItems");
+});
+
+it("should warn when @markAsPageable is applied to operation not returning model", async () => {
+  const diagnostics = await basicRunner.diagnose(`
+      @service
+      namespace TestService {
+        @Azure.ClientGenerator.Core.Legacy.markAsPageable
+        @route("/simple")
+        @get
+        op simpleOperation(): string;
+      }
+    `);
+
+  strictEqual(diagnostics.length, 1);
+  strictEqual(
+    diagnostics[0].code,
+    "@azure-tools/typespec-client-generator-core/invalid-mark-as-pageable-target",
+  );
+});
+
+it("should work with complex model return types", async () => {
+  await basicRunner.compile(`
+      @service
+      namespace TestService {
+        model PagedResult<T> {
+          @pageItems
+          value: T[];
+          @nextLink
+          nextLink?: string;
+        }
+
+        model Product {
+          id: string;
+          name: string;
+          price: float32;
+        }
+
+        @Azure.ClientGenerator.Core.Legacy.markAsPageable
+        @route("/products")
+        @get
+        op listProducts(): PagedResult<Product>;
+      }
+    `);
+
+  const methods = basicRunner.context.sdkPackage.clients[0].methods;
+  strictEqual(methods.length, 1);
+
+  const method = methods[0];
+  strictEqual(method.kind, "paging");
+  strictEqual(method.name, "listProducts");
+
+  const responseType = method.response.type;
+  ok(responseType);
+  strictEqual(responseType.kind, "array");
+});
+
+it("should work with operations returning list in different response structures", async () => {
+  await basicRunner.compile(`
+      @service
+      namespace TestService {
+        model ItemPage {
+          @pageItems
+          data: Item[];
+          @continuationToken
+          continuationToken?: string;
+        }
+
+        model Item {
+          id: string;
+          value: string;
+        }
+
+        @Azure.ClientGenerator.Core.Legacy.markAsPageable
+        @route("/items")
+        @get
+        op getItems(): ItemPage;
+      }
+    `);
+
+  const methods = basicRunner.context.sdkPackage.clients[0].methods;
+  strictEqual(methods.length, 1);
+
+  const method = methods[0];
+  strictEqual(method.kind, "paging");
+  strictEqual(method.name, "getItems");
+});
+
+it("should handle nested model responses", async () => {
```

Comment:
this looks redundant with the above test `should work with complex model return types`, therefore we could just remove this test.